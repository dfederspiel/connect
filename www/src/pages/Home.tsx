import { gql, useQuery } from '@apollo/client';
import { Typography } from '@material-ui/core';
import Affirmations from '../components/Affirmations/Affirmations';
import { Article } from '../generated/graphql';

const HomePage = () => {
  const { data } = useQuery<{ article: Article[] }>(gql`
    query {
      article {
        id
        title
        body
        user {
          id
          email
        }
      }
    }
  `);

  return (
    <>
      <Typography variant={'h2'}>Home</Typography>
      <Affirmations />
      {data &&
        data.article.map((a) => (
          <div key={`article_${a.id}`}>
            <h2>
              {a.title} by {a.user?.email}
            </h2>
            <p>{a.body}</p>
          </div>
        ))}
    </>
  );
};

export default HomePage;
