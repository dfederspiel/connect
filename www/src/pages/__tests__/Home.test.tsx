import { render } from '@testing-library/react';
import Home from '../Home';

describe('the home page', () => {
  it('renders as expected', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
