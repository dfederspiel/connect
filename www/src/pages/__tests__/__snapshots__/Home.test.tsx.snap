// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the home page renders as expected 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    Home
  </h2>
</DocumentFragment>
`;
